File: Aula5/Projeto.AspNet.05.BackEnd.WebAPI/Controllers/Data/Entity/Estudante.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace Projeto.AspNet._05.BackEnd.WebAPI.Controllers.Data.Entity
{
    
    public class Estudante
    {
        [Key]
        public int EstudanteId { get; set; }
        public string? EstudanteNome { get; set; }
        public string? EstudanteSobrenome { get; set; }
        public int? EstudanteRA { get; set; }
        public string? EstudanteEmail { get; set; }
        public int EstudanteIdade { get; set; }
        public string? EstudanteFone { get; set; }
        public string? EstudanteGenero { get; set; }
        public ICollection<Curso>? Curso { get; set; }
    }
}

File: Aula5/codigos-angular-front-end-asp-net/src/app/model/estudante.ts
// esta interface tem o propósito de atuar como model domain para a aplicação 

import { Curso } from "./curso";

export interface Estudante {
    //definir as props que serão utilizadas como "conjunto de regras" para a manipulação de dados
    estudanteId: number
    estudanteNome: string
    estudanteSobrenome: string
    estudanteRA: number;
    estudanteEmail: string;
    estudanteIdade: number;
    estudanteFone: string;
    estudanteGenero: string;
}
